### PR TITLE
[6] Simple enrichments

### LINKED ISSUE #2: Create enrichment utils (placeholders)
Create a `src/enrichment` folder and add some simple versions or placeholder functions for the data enrichments.
- Some of the enrichments will be specific to Crunchbase, and hence can be kept in their own file (eg crunchbase.py)
- Some enrichments might be more universal (eg classifying texts in terms of topics) - those could be in a separate file (eg topics.py)

### LINKED ISSUE #6: Add enrichments (simple versions)

### PR DESCRIPTION
**Pull request by beingkk**
Thank you @sqr00t for looking at this! The enrichment pipeline prototype is in 
`discovery_utils.src.crunchbase.py`

To test it out, you'll need to implement getters or download the data locally. 

Thanks!

### CODE REVIEW

File: src/enrichment/testing.ipynb
Comment (beingkk) [inline src/enrichment/testing.ipynb:1]: Hey @sqr00t please see below the prototype enrichment functions!

The main functions are `enrich_funding_rounds` and `enrich_organisations` which then call the other internal functions.

Note that there is a block of commented out CrunchbaseWrangler code in the beginning which I haven't used yet.

File: src/utils/google.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        try:
+            s3_client().download_file(
+                S3_BUCKET,
+                f"credentials/{credentials_json.name}",
+                str(credentials_json),
+            )
+        except Exception as e:
+            raise Exception(f"Error downloading credentials from S3: {e}")
+
+    return credentials_json
```
Comment (sqr00t) [inline src/utils/google.py:66]: This is a great way to share the credentials for gsheet!

File: src/enrichment/crunchbase.py [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+
+    return (
+        funding_rounds.query("id in @funding_round_ids")
+        # More informative column names
+        .rename(
+            columns={
+                "id": "funding_round_id",
+                "name": "funding_round_name",
+            }
+        )
```
Comment (beingkk) [inline src/enrichment/crunchbase.py:228]: FYI @sqr00t, here I'm renaming the `id` in funding_rounds to `funding_round_id` 
Just something to remember when looking at the diffs for funding_rounds table

File: pyproject.toml [hunk 3]
```diff
@@ -13,6 +13,15 @@ packages = [
 
 [tool.poetry.dependencies]
 python = ">=3.10, <3.13"
+boto3 = "*"
+urllib3 = "<2"
+pandas = "*"
+pyarrow = "^15.0.0"
+fastparquet = "^2024.2.0"
+currencyconverter = "^0.17.16"
```
Comment (sqr00t) [inline pyproject.toml]:
```suggestion
currencyconverter = "^0.17.16"
```
is there specific cases that fastparquet is needed? the newer updates on pandas would be leaning to pyarrow being default.

Comment (sqr00t) [reply pyproject.toml]: Just wondering if you've encountered any errors @beingkk . As the parquet raw crunchbase outputs are using pyarrow

Comment (beingkk) [reply pyproject.toml]: Hey - I think I probably had an error that suggested installing fastparquet, but I don't have strong feelings about this. Happy to use another approach if that makes more sense!

**Pull request merged**